src/accesspattern/dataiterator.jl

```diff
@@ -254,17 +254,17 @@ function RandomBatches{T,O}(data::T, size::Int, obsdim::O)
     RandomBatches{E,T,O,Base.IsInfinite}(data, size, 1337, obsdim)
 end
 
-RandomBatches(data, size::Int; obsdim = default_obsdim(data)) =
+RandomBatches{T}(data::T, size::Int; obsdim = default_obsdim(data)) =
     RandomBatches(data, size, obs_dim(obsdim))
 
-RandomBatches(data, size::Int, count::Int; obsdim=default_obsdim(data)) =
+RandomBatches{T}(data::T, size::Int, count::Int; obsdim = default_obsdim(data)) =
     RandomBatches(data, size, count, obs_dim(obsdim))
 
 # convenience constructor.
-RandomBatches(data, size::Int, ::Void, obsdim) =
+RandomBatches{T}(data::T, size::Int, ::Void, obsdim) =
```

Comment:
For the strangest reason. On my mackbook on 0.5 dispatch did not work as I would have assumed without it `RandomBatches(some_array, 10)` did not trigger https://github.com/JuliaML/MLDataUtils.jl/blob/329400a80685ef91a7971708aebbbd8ffe7b0453/src/accesspattern/dataiterator.jl#L138 but instead invoked https://github.com/JuliaML/MLDataUtils.jl/blob/329400a80685ef91a7971708aebbbd8ffe7b0453/src/accesspattern/dataiterator.jl#L133. I don't yet understand why